cmd/export/export.go

```diff
@@ -18,161 +16,48 @@ import (
-	Namespace string `mapstructure:"namespace"`
+	rawConfig              api.Config
+	exportDir              string
+	userSpecifiedNamespace string
 
-	//User Impersonation Flags
-	User  string   `mapstructure:"as-user"`
-	Group []string `mapstructure:"as-group"`
-	Extra string   `mapstructure:"as-extras"`
+	genericclioptions.IOStreams
 }
 
-func (o *ExportOptions) setExtras() error {
-	if o.Extra == "" {
-		return nil
+func (o *ExportOptions) Complete(c *cobra.Command, args []string) error {
+	var err error
+	o.rawConfig, err = o.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return err
 	}
-	keysAndStrings := strings.Split(o.Extra, ";")
-	o.extras = map[string][]string{}
-	for _, keysAndString := range keysAndStrings {
-		keyString := strings.Split(keysAndString, "=")
-		if len(keyString) != 2 {
-			// Todo: Much better error message here.
-			return fmt.Errorf("invalid extra options")
-		}
-		o.extras[keyString[0]] = strings.Split(keyString[1], ",")
+
+	o.userSpecifiedNamespace, _, err = o.configFlags.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
 	}
-	return nil
-}
 
-func (o *ExportOptions) Complete(c *cobra.Command, args []string) error {
-	// TODO: @alpatel
 	return nil
 }
 
 func (o *ExportOptions) Validate() error {
-	// TODO: @alpatel
-	if len(o.User) == 0 && len(o.Group) == 0 && len(o.Extra) != 0 {
-		return fmt.Errorf("if adding extras must also provide a group or user")
-	}
-
-	if err := o.setExtras(); err != nil {
-		return err
-	}
+	// This is where pre-run validations should be performed
 	return nil
 }
 
 func (o *ExportOptions) Run() error {
-	return o.run()
-}
-
-func NewExportCommand(streams genericclioptions.IOStreams, f *flags.GlobalFlags) *cobra.Command {
-	o := &ExportOptions{
-		configFlags: genericclioptions.NewConfigFlags(true),
-
-		IOStreams:        streams,
-		cobraGlobalFlags: f,
-	}
-	cmd := &cobra.Command{
-		Use:   "export",
-		Short: "Export the namespace resources in an output directory",
-		RunE: func(c *cobra.Command, args []string) error {
-			if err := o.Complete(c, args); err != nil {
-				return err
-			}
-			if err := o.Validate(); err != nil {
-				return err
-			}
-			if err := o.Run(); err != nil {
-				return err
-			}
-
-			return nil
-		},
-		PreRun: func(cmd *cobra.Command, args []string) {
-			viper.BindPFlags(cmd.Flags())
-			viper.Unmarshal(&o.Flags)
-			viper.Unmarshal(&o.globalFlags)
-		},
-	}
-
-	addFlagsForOptions(&o.cobraFlags, cmd)
-
-	return cmd
-}
-
-func addFlagsForOptions(o *Flags, cmd *cobra.Command) {
-	cmd.Flags().StringVarP(&o.ExportDir, "export-dir", "e", "export", "The path where files are to be exported")
-	cmd.Flags().StringVar(&o.Context, "context", "", "The kube context, if empty it will use the current context. If --namespace is set it will take precedence")
-	cmd.Flags().StringVarP(&o.Namespace, "namespace", "n", "", "The kube namespace to export.")
-	cmd.Flags().StringVar(&o.User, "as-user", "", "The user to impersonation.")
```

Comment:
Are we losing this ability with this PR?

Reply:
We are losing the `as-user` flag, but is this the same as `user` in `kubectl`? This is what our `--help` looks like with this change:

```
$ ./crane export --help
Export the namespace resources in an output directory

Usage:
  crane export [flags]

Flags:
      --as string                      Username to impersonate for the operation
      --as-group stringArray           Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
      --cache-dir string               Default cache directory (default "/home/dzager/.kube/cache")
      --certificate-authority string   Path to a cert file for the certificate authority
      --client-certificate string      Path to a client certificate file for TLS
      --client-key string              Path to a client key file for TLS
      --cluster string                 The name of the kubeconfig cluster to use
      --context string                 The name of the kubeconfig context to use
  -e, --export-dir string              The path where files are to be exported (default "export")
  -h, --help                           help for export
      --insecure-skip-tls-verify       If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
      --kubeconfig string              Path to the kubeconfig file to use for CLI requests.
  -n, --namespace string               If present, the namespace scope for this CLI request
      --request-timeout string         The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests. (default "0")
  -s, --server string                  The address and port of the Kubernetes API server
      --tls-server-name string         Server name to use for server certificate validation. If it is not provided, the hostname used to contact the server is used
      --token string                   Bearer token for authentication to the API server
      --user string                    The name of the kubeconfig user to use

Global Flags:
      --debug               Debug the command by printing more information
  -f, --flags-file string   Path to input file which contains a yaml representation of cli flags. Explicit flags take precedence over input file values.
```

Does `--user string                    The name of the kubeconfig user to use` give us what we need here?